src/samples/http/server-sse.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * @brief HTTP Server sse
+ *
+ * Sample server that return a response and keep it alive.  It
+ * implements server sent events. The server sends the data typed in
+ * stdin and the data is broadcasted to all the clients connected.
+ *
+ * To test it:
+ *
+ * run ./server-sse -p 8080
+ * open a browser in http://127.0.0.1:8080 or curl http://127.0.0.1:8080
+ * type something
+ *
+ * To see the usage help, -h or --help.
+ */
+
+
+#include <getopt.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "soletta.h"
+#include "sol-http.h"
+#include "sol-http-server.h"
+#include "sol-util.h"
+#include "sol-util-file.h"
+
+#define HTML_FILE \
+    "<!DOCTYPE html>" \
+    "<html>" \
+    "<body>" \
+    "<h1>Getting server updates</h1>" \
+    "<div id=\"result\"></div>" \
+    "<script>" \
+    "if(typeof(EventSource) !== \"undefined\") {" \
+    "var source = new EventSource(\"http://%.*s:%d/events\");" \
+    "source.onmessage = function(event) {" \
+    "document.getElementById(\"result\").innerHTML += event.data + \"<br>\";" \
+    "};" \
+    "} else {" \
+    "document.getElementById(\"result\").innerHTML = \"Sorry, your browser does not support server-sent events...\";" \
+    "}" \
+    "</script>" \
+    "</body>" \
+    "</html>"
+
+
+static struct sol_http_server *server;
+static struct sol_fd *stdin_watch;
+static struct sol_buffer value;
+static struct sol_ptr_vector responses = SOL_PTR_VECTOR_INIT;
+static int port = 8080;
+
+static bool
+on_stdin(void *data, int fd, uint32_t flags)
+{
+    uint16_t i;
+    struct sol_http_response_progressive *sse;
+    static const struct sol_str_slice begin = {
+        .len = sizeof("data: ") - 1,
+        .data = (const char *)"data: "
```

Comment:
nope, the macro is:
```
(struct sol_str_slice){.len = (_len), .data = (_s) }
```
This construction makes the semaphoreci's gcc complains about using not const data to initialize ..